apps/04TowersOfHanoi.js

```diff
@@ -18,25 +18,40 @@ function printStacks() {
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
-
-function movePiece() {
-  // Your code here
-
+// This code below enables the user to move the peices in the array
+function movePiece(startStack, endStack) {
+  stacks[endStack].push(stacks[startStack].pop());
 }
-
-function isLegal() {
+// This code below checks to see if the users move was legal
+function isLegal(startStack, endStack) {
   // Your code here
+  var startBlock = stacks[startStack][stacks[startStack].length - 1];
+  var endBlock = stacks[endStack][stacks[endStack].length - 1];
+  if (startBlock < endBlock || endBlock === undefined) {
+    movePiece(startStack, endStack);
+    return true;
+  } else if (startBlock > endBlock) {
+    console.log("Try Again");
+    return false;
 
-}
+  }
 
+}
+// This code below checks for a in - if the stack equals 4
 function checkForWin() {
   // Your code here
-
+  if (stacks.c.length === 4 || stacks.b.length === 4) {
+    return true;
+    console.log("You Won!!!");
```

Comment:
Can this console.log statement ever be reached? (It's right after a return statement)